### PR TITLE
Added import-check command, tests, and doc.

### PR DESCRIPTION
**Pull request by ggainey**
closes #7549

### CODE REVIEW

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py [hunk 1]
```diff
@@ -236,3 +298,118 @@ def test_chunked_import(self):
...
+                for s in result.toc.messages
+            )
+        )
+        # FAILS IN CI, passes locally
+        # self.assertTrue(
+        #     any(
+        #         "directory /tmp/importcheck/nowritedir must allow pulp write-access" in s
+        #         for s in result.toc.messages
+        #     )
+        # )
```
Comment (gerrod3) [inline pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:412]: You know why it doesn't work in the CI? I also got the test to pass locally.

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py [hunk 2]
```diff
@@ -236,3 +298,118 @@ def test_chunked_import(self):
...
+        result = self.import_check_api.pulp_import_check_post(body)
+        self.assertEqual(result.path.context, "/tmp/idonotexist")
+        self.assertFalse(result.path.is_valid)
+        self.assertTrue(
+            any("file /tmp/idonotexist does not exist" in s for s in result.path.messages)
+        )
+
+        body = {"toc": "/tmp/idonotexist"}
+        result = self.import_check_api.pulp_import_check_post(body)
+        self.assertEquals(result.toc.context, "/tmp/idonotexist")
```
Comment (gerrod3) [inline pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:361]: Pytest says assertEquals is deprecated.
```suggestion
        self.assertEqual(result.toc.context, "/tmp/idonotexist")
```

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py [hunk 3]
```diff
@@ -236,3 +298,118 @@ def test_chunked_import(self):
...
+        body = {"repo_mapping": '{"foo": "bar"'}
+        result = self.import_check_api.pulp_import_check_post(body)
+        self.assertEqual(result.repo_mapping.context, '{"foo": "bar"')
+        self.assertFalse(result.repo_mapping.is_valid)
+        self.assertEqual(result.repo_mapping.messages[0], "invalid JSON")
+
+    def test_import_check_not_allowed(self):
+        body = {"path": "/notinallowedimports"}
+        result = self.import_check_api.pulp_import_check_post(body)
+        self.assertEquals(result.path.context, "/notinallowedimports")
```
Comment (gerrod3) [inline pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:338]: Pytest says assertEquals is deprecated.
```suggestion
        self.assertEqual(result.path.context, "/notinallowedimports")
```

**Review by gerrod3 — COMMENTED** [review]
Looks good. BTW is the path argument always suppose to point to a tarball when specified? I pointed it to the toc file and the command returned with valid. Should there be a check that the file types for toc and path are json and tar.gz?

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py [hunk 4]
```diff
@@ -236,3 +298,118 @@ def test_chunked_import(self):
...
+        }
+        result = self.import_check_api.pulp_import_check_post(body)
+
+        self.assertFalse(result.path.is_valid)
+        self.assertEqual(len(result.path.messages), 1, "Only not-allowed should be returned")
+        self.assertEqual(result.path.messages[0], "/ is not an allowed import path")
+
+        self.assertFalse(result.toc.is_valid)
+        print(result.toc.messages)
+        self.assertTrue(
```
[the remark this answers is not in the document]
Comment (ggainey) [reply pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:400]: Indeed! will rmv

File: pulpcore/app/viewsets/importer.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 from django.http import Http404
 from django_filters.rest_framework import filters
+
 from drf_spectacular.utils import extend_schema
 from rest_framework import mixins
```
[the remark this answers is not in the document]
Comment (ggainey) [reply pulpcore/app/viewsets/importer.py:6]: That...is a good catch. No, it's a leftover when this was going to be a ViewSet and there was more code here. Good, that means this file doesn't need to be touched at all.

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py (hunk 3 above)
Comment (ggainey) [reply pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:338]: Yeah, looks like I missed a couple of those - been using assertEquals() for too long.

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py (hunk 1 above)
Comment (ggainey) [reply pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:412]: I haven't determined why the ci-task thinks that it has write-access to a directory with 755 perms. I only commented this out after four different attempts, decided not to block on it.

**Comment by ggainey** [comment]
> Looks good. BTW is the path argument always suppose to point to a tarball when specified? I pointed it to the toc file and the command returned with valid. Should there be a check that the file types for toc and path are json and tar.gz?

I can't rely on the filename, because users can call the files whatever they want. And I don't want to add heuristics to open the files to look inside, this is supposed to be a top-level sanity-check - are the files in a location we're allowed to read from? do they exist? can I read them? are the directory-permissions correct (where possible)? Anything more than that, will be found when one actually does the import (and there are A LOT more checks there)

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py (hunk 1 above)
Comment (daviddavis) [reply pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:412]: You could use tmate to debug: https://github.com/marketplace/actions/debugging-with-tmate

Just add this to the end of [the ci workflow](https://github.com/pulp/pulpcore/blob/master/.github/workflows/ci.yml) and uncomment this test:

```yaml
    - name: Setup tmate session
      uses: mxschmitt/action-tmate@v3
      if: failure()
```

**Review by daviddavis — APPROVED** [review]
LGTM.

File: docs/workflows/import-export.rst [hunk 6]
```diff
@@ -364,3 +364,30 @@ The command to create an import will return a task that can be used to monitor t
...
+
+    * paths are in ``ALLOWED_IMPORT_PATHS``
+    * containing directory exists
+    * containing directory is readable
+    * path/toc file(s) exist and are readable
+    * for TOC, containing directory is writeable
+    * repo_mapping is valid JSON
+
+``import-check`` is a low-overhead synchronous call. It does make any attempt to do validations that
+require database access or long-running tasks such as verifying checksums. All parameters are optional.
```
Comment (goosemania) [inline docs/workflows/import-export.rst:388]: I'm not sure I fully follow the sentence. Did you mean "it does make no attempt", or "it does not make any attempt" or something like that?

File: pulpcore/app/serializers/__init__.py [hunk 7]
```diff
@@ -53,6 +53,9 @@
     ImporterSerializer,
     PulpImporterSerializer,
     PulpImportSerializer,
+    PulpImportCheckSerializer,
+    PulpImportCheckResponseSerializer,
+    EvaluationSerializer,
```
Comment (goosemania) [inline pulpcore/app/serializers/__init__.py:58]: Alphabetical order, please

File: pulpcore/app/serializers/importer.py [hunk 8]
```diff
@@ -170,3 +166,76 @@ class Meta:
...
+    create an importer.
+    """
+
+    path = serializers.CharField(
+        help_text=_("Path to export-tar-gz that will be imported."), required=False
+    )
+    toc = serializers.CharField(
+        help_text=_(
+            "Path to a table-of-contents file describing chunks to be validated, "
+            + "reassembled, and imported."
```
Comment (goosemania) [inline pulpcore/app/serializers/importer.py:221]: Not important but I don't think you need `+` here

File: pulpcore/app/views/importer.py [hunk 9]
```diff
@@ -0,0 +1,124 @@
+from gettext import gettext as _
+import json
+import os
+from drf_spectacular.utils import extend_schema
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from pulpcore.app import settings
+from pulpcore.app.serializers import PulpImportCheckSerializer, PulpImportCheckResponseSerializer
```
Comment (goosemania) [inline pulpcore/app/views/importer.py:9]: mega nitpick, in theory those imports should be in different order ;)

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py [hunk 10]
```diff
@@ -25,6 +26,7 @@
     ExportersCoreExportsApi,
     ImportersPulpApi,
     ImportersCoreImportsApi,
+    ImportersCoreImportCheckApi,
```
Comment (goosemania) [inline pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:29]: I don't know why I notice those order things, don't ask me :)
One is not from your PR, to be fair

**Review by goosemania — APPROVED** [review]
Thanks for this feature!
I added just some minor comments.  It's up to you, whether to address them or not.

File: docs/workflows/import-export.rst (hunk 6 above)
Comment (ggainey) [reply docs/workflows/import-export.rst]: great catch, fixed

File: pulpcore/app/serializers/importer.py (hunk 8 above)
Comment (ggainey) [reply pulpcore/app/serializers/importer.py]: Embarassing, my Java history shining thru...

File: pulpcore/app/serializers/__init__.py (hunk 7 above)
Comment (ggainey) [reply pulpcore/app/serializers/__init__.py]: reordered a number of imports, just for you :)

**Pull request merged**